src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,86 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException; // Правильний імпорт винятку
+import core.basesyntax.model.User;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RegistrationServiceTest {
+    private RegistrationService registrationService;
+    private StorageDaoImpl storage;
+
+    @Before
+    public void setUp() {
+        storage = new StorageDaoImpl(); // Використовуємо справжню реалізацію
+        registrationService = new RegistrationServiceImpl(storage);
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User validUser = new User("validLogin", "validPassword", 20);
+        registrationService.register(validUser);
+        assertEquals(validUser, storage.get(validUser.getLogin()));
```

Comment:
According to the checklist, you should use low-level methods like `Storage.people.add()` directly instead of `RegistrationService.register()` to add users for testing. This ensures that the tests do not rely on the method under test for setting up test conditions.